datashader/transfer_functions/__init__.py

```diff
@@ -391,50 +393,71 @@ def _colorize(agg, color_key, how, alpha, span, min_alpha, name, color_baseline,
     data = agg_t.data.transpose([1, 2, 0])
     if da and isinstance(data, da.Array):
         data = data.compute()
-    color_data = data.copy()
+
+    H, W, C = data.shape
+    color_data = ascontigous(data.copy())
```

Comment:
```suggestion
    color_data = ascontigous(data)
    if color_data is data:
    	color_data = color_data.copy()
```

No need for a potential double copy.

``` python
import numpy as np

x = np.asfortranarray(np.zeros((10, 10)))
np.ascontiguousarray(x) is x
```

We could also look into something like `x.flags['C_CONTIGUOUS']`